cvat/apps/documentation/user_guide.md

```diff
@@ -1564,70 +1557,86 @@ Shapes that don't have `group_id`, will be highlighted in white.
+
+![](static/documentation/images/image205.jpg)
+
+You can add multiple rules, to do so click the add rule button and set another rule. Once you've set a new rule, you'll be able to choose which operator they will be connected by: `And` or `Or`.
+
+![](static/documentation/images/image206.jpg)
+
+All subsequent rules will be joined by the chosen operator. Click `Submit` to apply the filter or if you want multiple rules to be connected by different operators, use groups.
+
+### Groups
+
+To add a group, click the "add group" button. Inside the group you can create rules or groups.
+
+![](static/documentation/images/image207.jpg)
+
+If there is more than one rule in the group, they can be connected by `And` or `Or` operators.
+The rule group will work as well as a separate rule outside the group and will be joined by an
+ operator outside the group.
+You can create groups within other groups, to do so you need to click the add group button within the group.
+
+You can move rules and groups. To move the rule or group, drag it by the button.
+To remove the rule or group, click on the `Delete` button.
+
+![](static/documentation/images/image208.jpg)
+
+If you activate the `Not` button, objects that don't match the group will be filtered out.
+Click `Submit` to apply the filter.
+The "Cancel" button undoes the filter. The `Clear filter` button removes the filter.
 
-**Examples filters**
-
-- `label=="car" | label==["road sign"]` - this filter will show only objects with the car or road sign label.
-- `shape == "polygon"` - this filter will show only polygons.
-- `width >= height` - this filter will show only those objects whose width will be greater than
-  or equal to the height.
-- `attr["color"] == "black"` - this filter will show objects whose color attribute is black.
-- `clientID == 50` - this filter will show the object with id equal to 50 (e.g. rectangle 50).
-- `(label=="car" & attr["parked"]==true) | (label=="pedestrian" & width > 150)` - this filter will display objects
-  with the “car” label and the parking attribute enabled or objects with the “pedestrian” label with a height of more
-  than 150 pixels
-- `(( label==["car \"mazda\""]) | (attr["parked"]==true & width > 150)) & (height > 150 & (clientID == serverID)))` -
-  This filter will show objects with the label "car" mazda "" or objects that have the parked attribute turned on
-  and have a width of more than 150 pixels, and those listed should have a height of more than 150 pixels
-  and their clientID is equal to serverID.
-
-**Filter history**
-
-![](static/documentation/images/image175.jpg)
-
-You can add previously entered filters and combine them. To do so, click on the input field and a list of previously
-entered filters will open. Click on the filters to add them to the input field.
-Combined filters occur with the "or" operator.
+After the filter is applied it will automatically appear in the `Recently used` list.
```

Comment:
Once applied filter automatically appears in "Recent used" list. Maximum length of the list is 10.